### DISCUSSION TITLE
Add support for dynamically hiding tabs in Overflow if they do not fit the viewport

### DISCUSSION THREAD

**Discussion by unknownzone96** [vmware/clarity#5939] [General]
Dynamically hiding the tabs that do not fit the viewport in the Overflow dropdown is not currently supported: https://clarity.design/angular-components/tab/#overflow
Additional code is required to make Clarity Overflow responsive.
IMHO it is a common practice to dynamically add tabs in overflow when sizing down the viewport, hiding rightmost tab first.

**Comment by Jinnie** [comment]
For performance reasons Clarity will not listen to viewport changes. If it is important for you, you can implement it on your side.
Just keep in mind that the overflow pattern is very controversial from A11Y point of view and not really recommended by our Accessibility team. It will not be present in the Clarity Core tabs.

**Reply by mattmutt** [reply]
why can't UI design support a use case where rearranging tabs is a desirable pattern?

Good responsive design can be inclusive of accessibility concerns and tackle the age-old problem of how do components deal with their layout, responsibly.

Other design philosophies indeed support the idea of revealing what is possible in the viewport and then transparently shifting the other pieces that would be invisible anyways into that "..." menu drop down.

Intuitive, clean, effective without needless scrolling and guessing. I would recommend the design team soliciting more external feedback if the above points are not agreeable.

**Reply by mathisscott** [reply]
@mattmutt 

I think the issue there is designing one solution that meets all possible use cases. This is why the Core components have been working to be as non-intrusive as possible. Years of working away at this have taught me that flexibility and customization are more important than any sort of config-it-and-go componentry.

In addition, we have a lot of products that we support. And each one of them has their own UX team and their own a11y contact. Many of these designers and a11y folks have different requirements on a feature like this based on their product's use cases. There's not a "one size fits all" use case with something like this. And that's before we take into account the companies using Clarity outside of VMware that have their own use cases.

**Reply by Jinnie** [reply]
Also, as this can be achieved on the user side, we're open to receiving feedback, as patterns and samples that achieve this or any other scenario which remains beyond our scope. At the moment we track such patterns and samples internally as part of our support tools and processes. If we accumulate enough valuable information, we may use it as a basis for a more interactive pattern sharing solution.
